prow/plank/controller.go

```diff
@@ -136,27 +138,28 @@ func NewController(kc, pkc *kube.Client, ghc *github.Client, ca *config.Agent, t
 		ca:          ca,
 		node:        n,
 		pendingJobs: make(map[string]int),
-		lock:        sync.RWMutex{},
 		totURL:      totURL,
+		selector:    selector,
 	}, nil
 }
 
 // Sync does one sync iteration.
 func (c *Controller) Sync() error {
-	pjs, err := c.kc.ListProwJobs(kube.EmptySelector)
+	pjs, err := c.kc.ListProwJobs(c.selector)
 	if err != nil {
 		return fmt.Errorf("error listing prow jobs: %v", err)
 	}
-	selector := fmt.Sprintf("%s = %s", kube.CreatedByProw, "true")
-	pods, err := c.pkc.ListPods(selector)
+	selectOwned := fmt.Sprintf("%s = %s", kube.CreatedByProw, "true")
+	pods, err := c.pkc.ListPods(c.selector + "," + selectOwned)
```

Comment:
This won't be a valid selector if `c.selector` is `kube.EmptySelector`.

Reply:
Oops looks like this was never fixed. I'll open a PR.

Reply:
ref #5387

Reply:
uh, so I just realized today that `plank` was the only part of prow that wasn't redeployed during my last upgrade because I was fiddling with it and had disabled its image change trigger (an openshift concept we use to redeploy on image changes).